### DISCUSSION TITLE
UI question

### DISCUSSION THREAD

**Discussion by Fractal-Tess** [pocketbase/pocketbase#1320] [Q&A] [answered]
Hey there,

Would you mind describing how the WebUI works?
I've been looking to create something like this for myself using rust and svelte, but I'm lost on how you turn a SPA(svelte) into an experience that feels like there is a server below it.

AFAIK, SPAs require you to go to the index page in order to get yourself the needed javascript and HTML file, and from there on out, a client-side router kicks in and you are free as you wish. Yet with Pocketbase, the experience is much nicer, allowing you to go straight to your desired page without having to start off from the index page.

From what I can gather, it seems like you are using the `/_/` path and then some query parameters for the route itself.
I would love to do something like this myself using rust, so it would be awesome if would give me any pointers for that.

Cheers!

**Comment by ganigeorgiev** [comment, marked as answer]
`/_/` is just a path that serves the index.html from the [ui/dist folder](https://github.com/pocketbase/pocketbase/tree/master/ui/dist)

We use a hash-based router (eg. `/#/some-route`) meaning that `/some-route` is not an actual path but a part of the url hash/fragment (the accessed url is actually `index.html#/some-route`).

If you are working on a plain Svelte application you can check https://github.com/ItalyPaleAle/svelte-spa-router or other similar routers.

If you are using SvelteKit, then you may want to enable the [`trailingSlash` option of `adapter-static`](https://github.com/sveltejs/kit/tree/master/packages/adapter-static) (this will generate a directory with index.html for each of your routes, aka. `/some-route/index.html`, which most hosts usually would render on `/some-route/` access).

**Reply by Fractal-Tess** [reply]
Thanks for the answer,

I only ever knew that react had a router that worked with hash-based routing, but I actually never thought about exploring options and solutions with this hash-based routing.
